content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,284 @@
+conflict_prefer("filter", "dplyr", quiet = TRUE)
+out <- lineitem |>
+  select(l_shipdate, l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  filter(l_shipdate <= !!as.Date("1998-09-02")) |>
+  select(l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  summarise(
+    sum_qty = sum(l_quantity),
+    sum_base_price = sum(l_extendedprice),
+    sum_disc_price = sum(l_extendedprice * (1 - l_discount)),
+    sum_charge = sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)),
+    avg_qty = mean(l_quantity),
+    avg_price = mean(l_extendedprice),
+    avg_disc = mean(l_discount),
+    count_order = n(),
+    .by = c(l_returnflag, l_linestatus)
+  ) |>
+  arrange(l_returnflag, l_linestatus)
+out
+```
+
+Like with other dplyr backends like dtplyr and dbplyr, duckplyr allows you to get faster results without learning a different syntax.
+Unlike other dplyr backends, duckplyr does not require you to change existing code or learn specific idiosyncracies.
+Not only is the syntax the same, the semantics are too!
+
+The duckplyr package is fully compatible with dplyr: if an operation cannot be carried out with DuckDB, it is automatically outsourced to dplyr.
+Over time, we expect fewer and fewer fallbacks to dplyr to be needed.
+
+
+## How to use duckplyr
+
+To _replace_ dplyr with duckplyr, you can:
+
+- Load duckplyr and then keep your pipeline as is. Calling `library(duckplyr)` overwrites dplyr methods, enabling duckplyr for the entire session no matter how data.frames are created.
+    This is shown in the example above.
+
+- Create individual "duck frames" using _conversion functions_ like `duckdb_tibble()` or `as_duckdb_tibble()`, or _ingestion functions_ like `read_csv_duckdb()`.
+
+
+Then, the data manipulation pipeline uses the exact same syntax as a dplyr pipeline.
+The duckplyr package performs the computation using DuckDB.
+
+```{r, echo=FALSE, message=FALSE, warning=FALSE}
+# Undo the effect of library(duckplyr)
+methods_restore()
+```
+
+
+```{r}
+out <- lineitem |>
+  duckplyr::as_duckdb_tibble() |> # this is the only change :-)
+  select(l_shipdate, l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  filter(l_shipdate <= !!as.Date("1998-09-02")) |>
+  select(l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  summarise(
+    sum_qty = sum(l_quantity),
+    sum_base_price = sum(l_extendedprice),
+    sum_disc_price = sum(l_extendedprice * (1 - l_discount)),
+    sum_charge = sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)),
+    avg_qty = mean(l_quantity),
+    avg_price = mean(l_extendedprice),
+    avg_disc = mean(l_discount),
+    count_order = n(),
+    .by = c(l_returnflag, l_linestatus)
+  ) |>
+  arrange(l_returnflag, l_linestatus)
+```
+
+The result can finally be materialized to memory, or computed temporarily, or computed to a file.
+
+```{r}
+# to memory
+nrow(out)
+# or for instance collect(out)
```

Comment:
How and why does `nrow()` materialize to memory?

I had this in a separate paragraph in my edits.